tests/next_tests/integration_tests/feature_tests/cases.py

```diff
@@ -0,0 +1,358 @@
+                dtype=dtype or arg_type.dtype.kind.name.lower(),
+            )
+        case ts.ScalarType():
+            strategy = strategy or scalar5
+            return strategy(dtype=dtype or arg_type.kind.name.lower(), shape=arg_type.shape)
+        case ts.TupleType():
+            return tuple(
+                (
+                    _allocate_for(
+                        case=case, arg_type=t, sizes=sizes, dtype=dtype, strategy=strategy
+                    )
+                    for t in arg_type.types
+                )
+            )
+        case _:
+            raise TypeError(f"Can not allocate for type {arg_type}")
+
+
+def run(
+    case: Case,
+    fieldview_prog: decorator.FieldOperator | decorator.Program,
+    *args: common.Field,
+    **kwargs: Any,
+) -> None:
+    """Run fieldview code in the context of a given test case."""
+    if kwargs.get("offset_provider", None) is None:
+        kwargs["offset_provider"] = case.offset_provider
+    fieldview_prog.with_backend(case.backend)(*args, **kwargs)
+
+
+def get_default_data(
+    case: Case,
+    fieldview_prog: decorator.FieldOperator | decorator.Program,
+) -> tuple[tuple[common.Field | int | float, ...], dict[str : common.Field | int | float]]:
+    """
+    Allocate default params and return values for a fieldview code object given a test case.
+
+    Meant to reduce boiler plate for simple cases, everything else should rely on `allocate()`.
+    """
+    param_types = get_param_types(fieldview_prog)
+    kwfields: dict[str, Any] = {}
+    if not isinstance(param_types.setdefault(RETURN, types.NoneType), types.NoneType):
+        kwfields = {"out": allocate(case, fieldview_prog, RETURN).strategy(zeros)()}
+    param_types.pop(RETURN)
+    inps = tuple(
+        allocate(case, fieldview_prog, name)()
+        for name in get_param_types(fieldview_prog)
+        if name != "out" and name != RETURN
+    )
+    return inps, kwfields
+
+
+def verify(
+    case: Case,
+    fieldview_prog: decorator.FieldOperator | decorator.Program,
+    *args: common.Field,
+    out: Optional[common.Field] = None,
+    nopass_out: Optional[common.Field] = None,
+    ref: common.Field,
+    offset_provider: Optional[dict[str, common.Connectivty | common.Dimension]] = None,
+    comparison: Callable[[Any, Any], bool] = np.allclose,
+) -> None:
+    """
+    Check the result of executing a fieldview program or operator against ref.
+
+    One of `out` or `nopass_out` must be passed.
+    If `out` is passed it will be used as an argument to the fieldview program and compared against `ref`.
+    Else, `nopass_out` will not be passed and compared to `ref`.
+    """
+    if out:
+        run(
+            case,
+            fieldview_prog,
+            *args,
+            out=out,
+            offset_provider=offset_provider,
+        )
+    else:
+        run(case, fieldview_prog, *args, offset_provider=offset_provider)
+
+    assert comparison(ref, out or nopass_out)
```

Comment:
I don't understand how the `nopass_out` argument works here...

Reply:
I renamed it and documented it (hopefully) more specifically in the docstring. Is it more understandable now?